rbc/irtools.py

```diff
@@ -338,31 +380,87 @@ def compile_to_LLVM(functions_and_signatures,
+        for fid, sig in signatures.items():
             fname = func.__name__ + sig.mangling()
-            function_names.append(fname)
             args, return_type = sigutils.normalize_signature(
                 sig.tonumba(bool_is_int8=True))
-            cres = compiler.compile_extra(typingctx=typing_context,
-                                          targetctx=target_context,
-                                          func=func,
-                                          args=args,
-                                          return_type=return_type,
-                                          flags=flags,
-                                          library=main_library,
-                                          locals={},
-                                          pipeline_class=pipeline_class)
+            try:
+                cres = compiler.compile_extra(typingctx=typing_context,
+                                              targetctx=target_context,
+                                              func=func,
+                                              args=args,
+                                              return_type=return_type,
+                                              flags=flags,
+                                              library=main_library,
+                                              locals={},
+                                              pipeline_class=pipeline_class)
+            except OmnisciUnsupportedError as msg:
+                for m in re.finditer(r'[|]OmnisciUnsupportedError[|](.*?)\n', str(msg), re.S):
+                    print(f'Skipping {fname} `{sig}`: {m.group(0)[25:]}')
+                continue
+            except nb_errors.TypingError as msg:
+                for m in re.finditer(r'[|]OmnisciUnsupportedError[|](.*?)\n', str(msg), re.S):
+                    print(f'Skipping {fname} `{sig}`: {m.group(0)[25:]}')
+                    break
+                else:
+                    raise
+                continue
+            except Exception:
+                raise
+
+            skip = False
+            intrinsics, declarations, defined, libraries = get_called_functions(
+                cres.library, cres.fndesc.llvm_func_name)
+            for f in declarations:
+                if target.is_gpu:
+                    if f.startswith('__nv_') and f[5:] in get_libdevice_functions():
+                        # numba supports libdevice functions starting from 0.52
+                        continue
+
+                if target.is_cpu:
+                    if f in get_libm_functions():
+                        # omniscidb is linked against m library
+                        continue
+                    if f in get_stdio_functions():
+                        # omniscidb is linked against stdio library
+                        continue
+                    if f in get_stdlib_functions():
+                        # omniscidb is linked against stdlib library
+                        continue
+
```

Comment:
Move this part to omniscidb.py?

Reply:
Done. This part is now in omniscidb.py where the TargetInfo instances are created.